hack/docs/instancetypes_gen_docs.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter/pkg/cloudprovider"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/utils/resources"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"log"
+	"os"
+	"sort"
+	"strings"
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 2 {
+		log.Printf("Usage: %s subnet-discovery-tag-value path/to/markdown.md", os.Args[0])
+		os.Exit(1)
+	}
+
+	os.Setenv("AWS_SDK_LOAD_CONFIG", "true")
+	ctx := context.Background()
+
+	cp := aws.NewCloudProvider(ctx, cloudprovider.Options{
+		ClientSet:  nil,
+		KubeClient: nil,
+	})
+	provider := v1alpha1.AWS{SubnetSelector: map[string]string{
+		"karpenter.sh/discovery": flag.Arg(0),
+	}}
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	if err := enc.Encode(provider); err != nil {
+		log.Fatalf("encoding provider, %s", err)
+	}
+	instanceTypes, err := cp.GetInstanceTypes(ctx, &v1alpha5.Provider{
+		Raw:    buf.Bytes(),
+		Object: nil,
+	})
+	if err != nil {
+		log.Fatalf("listing instance types, %s", err)
+	}
+
+	outputFileName := flag.Arg(1)
+	f, err := os.Create(outputFileName)
+	if err != nil {
+		log.Fatalf("error creating output file %s, %s", outputFileName, err)
+	}
+
+	log.Println("writing output to", outputFileName)
+	fmt.Fprintf(f, `---
+title: "Instance Types"
+linkTitle: "Instance Types"
+weight: 100
+
+description: >
+  Evaluate Instance Type Resources
+---
+`)
+	fmt.Fprintln(f, "<!-- this document is generated from hack/docs/instancetypes_gen_docs.go -->")
+	fmt.Fprintln(f, `AWS instance types offer varying resources and can be selected by labels. In particular the
+ephemeral-storage is the default value for an AL2 family assuming no specific block device mappings have been configured.`)
+
```

Comment:
Might be worth calling out the the resources have overhead taken out.  Although it looks like ephemeral storage does not and maybe not CPU either?  https://github.com/aws/karpenter/blob/205c4db5743f1f6824253e92304a5df2f875c90c/pkg/cloudprovider/aws/amifamily/al2.go#L94

Might also be worth calling out that pods are assuming the `aws-eni-limited-pod-density = true`

Reply:
SGTM